packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/android/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,447 @@
+
+This code shows the `handleResponse` function that handles the an SDK response indicating a next step and includes all the types of remediations that are part of a Google Authenticator flow:
+
+```kotlin
+private suspend fun handleResponse(response: IdxResponse) {
+    // If a response is successful, immediately exchange it for a token and set it on LoggedInUserView
+    if (response.isLoginSuccessful) {
+        when (val exchangeCodesResult = client?.exchangeInteractionCodeForTokens(response.remediations[ISSUE]!!)) {
+            is IdxClientResult.Error -> {
+                // handle error
+            }
+            is IdxClientResult.Success -> {
+               // handle success and get tokens from `exchangeCodesResult.result`
+            }
+        }
+        return
+    }
+
+    // Check for messages, such as entering an incorrect code or auth error and abort if there is message.
+    if (response.messages.size != 0) {
+        // handle error
+        return
+    }
+
+    // If no remediations are present, abort the login process and show error
+    if (response.remediations.size == 0) {
+        // handle error
+        return
+    }
+
+    // Handle the different sign-in steps (remediations) for a policy.
+    val remediation = response.remediations.first()
+    // check if there is a skip action
+    val skipRemediation = response.remediations.find { it.type == SKIP }
+
+    when (remediation.type) {
+        // Username and password, though password may be separate; see the next case.
+        IDENTIFY -> handleIdentify(remediation)
+        // Request a password or a passcode, such as one from Google Authenticator.
+        // Identity-first policies request the password separately.
+        CHALLENGE_AUTHENTICATOR -> handleChallenge(remediation)
+        // Display a list of authenticators to select or Enroll in an Authenticator
+        SELECT_AUTHENTICATOR_ENROLL, ENROLL_AUTHENTICATOR -> handleAuthenticatorEnrollOrChallenge(remediation, skipRemediation)
+        else -> {
+            // handle error
+        }
+    }
+}
+```
+
+The `handleIdentify()` function updates the username and password fields in the remediation object, and then calls `client.proceed(remediation)` which both sends the updated remediation to the server and requests the next step.
+
+```kotlin
+private fun handleIdentify(remediation: IdxRemediation) {
+    // Update the values in the remediation object and proceed to the next step in IDX flow
+    remediation["identifier"]?.value = username
+    remediation["credentials.passcode"]?.value = password
+    remediation.proceed()
+}
+```
+
+You can create `proceed()` as an extension function for `IdxRemediation` so that you can just call `remediation.proceed()` and reuse the code for updating the remediation and sending it to the server.
```

Comment:
While we _can_ do this. I would advocate for not doing it.

If we are going to do it, we should move it above the first call site.

I think we shouldn't do it because it makes it unclear where this is available and where it isn't. If we always use client.proceed(remediation), it's clear what's needed, and when it's available. Making the docs more clear.

Reply:
this is the first call site in the guide. I did it for the purpose of reusability as the same is required in multiple places and to be fair the extension mechanism is used through out the IDX kotlin samples so I just followed the same. I can change it to a normal function instead if it makes it clearer

Reply:
Extension functions are powerful, but I think in a guide that doesn't have the full context, and jump to definition is surely going to be confusing. 

In the sample I wrote, it's not an extension function: https://github.com/okta/okta-idx-android/blob/master/dynamic-app/src/main/java/com/okta/idx/android/dynamic/auth/DynamicAuthViewModel.kt#L341

Reply:
In that case it would make sense to not use an extension functions at all in guides and I think that could be helpful; for novice kotlin users as well, so gonna make all these simple functions